aicssegmentation/workflow/workflow.py

```diff
@@ -162,3 +169,148 @@ def is_done(self) -> bool:
             (bool): True if all WorkflowSteps have been executed, False if not
         """
         return self._next_step >= len(self._definition.steps)
+
+
+class BatchWorkflow:
+    """
+    Represents a batch of workflows to process.
+    This class provides the functionality to run batches of workflows using multiple image inputs from a input directory
+    according to the steps defined in its WorkflowDefinition.
+    """
+
+    def __init__(
+        self, workflow_definition: WorkflowDefinition, input_dir: str, output_dir: str, channel_index: int = 0
+    ):
+        if workflow_definition is None:
+            raise ValueError("workflow_definition")
+        self._workflow_definition = workflow_definition
+
+        self.input_path = Path(input_dir)
+        if not self.input_path.exists():
+            raise ValueError("The input directory does not exist")
+
+        self.output_path = Path(output_dir)
+        # Creating an the output directory at output_dir if it does not exist already
+        if not self.output_path.exists():
+            if self.output_path.parent.exists():
+                self.output_path.mkdir(parents=True, exist_ok=True)
+            else:
+                raise ValueError(f"Output directory does not exist, and cannot be created at {self.output_path.parent}")
+
+        self._files_count: int = 0
+        self._failed_files: int = 0
+        self._channel_index = channel_index
+        self._log_file: Path = self.output_path.joinpath("log.txt")
```

Comment:
what if there is already a log file in the directory? should we add a existence check and warn users if log exists?

Reply:
changed it so a new log file is generated every time